sidecar/internal/csiaddonsnode/csiaddonsnode.go

```diff
@@ -198,14 +204,28 @@ func (mgr *Manager) getCSIAddonsNode() (*csiaddonsv1alpha1.CSIAddonsNode, error)
 		if len(rs.OwnerReferences) == 0 {
 			return nil, fmt.Errorf("%w: replicaset has no owner", errInvalidConfig)
 		}
-		ownerReferences = append(ownerReferences, rs.OwnerReferences[0])
+		ownerKindForCSIAddonsName = rs.OwnerReferences[0].Kind
+		ownerNameForCSIAddonsName = rs.OwnerReferences[0].Name
+
+		// The pod might move to new nodes and this might create the stale CSIAddonsNode object.
+		// So, we need to set the pod as the owner for the deployment as we dont store any details
```

Comment:
I don't understand this comment. `set the pod as the owner for the deployment`... Which deployment? The Deployment creates a ReplicaSet which creates the Pods :thinking:

Looking at the code, you mean `set the pod as the owner for the CSIAddonsNode` instead.

Reply:
Yes thats correct, updated the comment to refect the same where we need to set pod as the owner to avoid garbage connections and CR's